Code/Framework/AzToolsFramework/AzToolsFramework/ComponentModes/BaseShapeComponentMode.cpp

```diff
@@ -0,0 +1,369 @@
+            ActionProperties actionProperties;
+            actionProperties.m_name = "Reset Current Mode";
+            actionProperties.m_description = "Reset Current Mode";
+            actionProperties.m_category = category;
+
+            actionManagerInterface->RegisterAction(
+                EditorMainWindowActionContextIdentifier,
+                actionIdentifier,
+                actionProperties,
+                []
+                {
+                    auto componentModeCollectionInterface = AZ::Interface<ComponentModeCollectionInterface>::Get();
+                    AZ_Assert(componentModeCollectionInterface, "Could not retrieve component mode collection.");
+
+                    componentModeCollectionInterface->EnumerateActiveComponents(
+                        [](const AZ::EntityComponentIdPair& entityComponentIdPair, const AZ::Uuid&)
+                        {
+                            ShapeComponentModeRequestBus::Event(entityComponentIdPair, &ShapeComponentModeRequests::ResetShapeSubMode);
+                        });
+                });
+
+            hotKeyManagerInterface->SetActionHotKey(actionIdentifier, "R");
+        }
+    }
+
+    void BaseShapeComponentMode::BindActionsToModes(const char* shapeName, const char* className)
+    {
+        auto actionManagerInterface = AZ::Interface<ActionManagerInterface>::Get();
+        AZ_Assert(actionManagerInterface, "BaseShapeComponentMode - could not get ActionManagerInterface on RegisterActions.");
+
+        AZStd::string modeIdentifier = AZStd::string::format("o3de.context.mode.%s", className);
+
+        actionManagerInterface->AssignModeToAction(
+            modeIdentifier, AZStd::string::format("o3de.action.%sComponentMode.setDimensionsSubMode", shapeName));
+        actionManagerInterface->AssignModeToAction(
+            modeIdentifier, AZStd::string::format("o3de.action.%sComponentMode.setTranslationOffsetSubMode", shapeName));
+        actionManagerInterface->AssignModeToAction(
+            modeIdentifier, AZStd::string::format("o3de.action.%sComponentMode.resetCurrentMode", shapeName));
+    }
+
+    void BaseShapeComponentMode::BindActionsToMenus(const char* shapeName)
+    {
+        auto menuManagerInterface = AZ::Interface<MenuManagerInterface>::Get();
+        AZ_Assert(menuManagerInterface, "BaseShapeComponentMode - could not get MenuManagerInterface on BindActionsToMenus.");
+
+        menuManagerInterface->AddActionToMenu(
+            EditMenuIdentifier, AZStd::string::format("o3de.action.%sComponentMode.setDimensionsSubMode", shapeName), 6000);
+        menuManagerInterface->AddActionToMenu(
+            EditMenuIdentifier, AZStd::string::format("o3de.action.%sComponentMode.setTranslationOffsetSubMode", shapeName), 6001);
+        menuManagerInterface->AddActionToMenu(
+            EditMenuIdentifier, AZStd::string::format("o3de.action.%sComponentMode.resetCurrentMode", shapeName), 6002);
```

Comment:
nit: maybe it'd be worth caching the component first one and then just add the final part in each case. I think it'd be more readable since it's not a new string to check each time and less errors to make. Same in other functions.

Reply:
Updated